### PR TITLE
Creating intentdiagram

### PR DESCRIPTION
**Pull request by WeissSa**
## Pull Request Summary
Added a semi-styled IntentDiagram component meant to be used to list the various questions intent by intent. Also made small edit to MainPage.jsx testing.

### Change types

- [x] Non-breaking change
- [x] Adding tests
- [x] Front end change

### Testing
- [x] I have added testing to this PR 

### How to manually test this PR (QA instructions)
Go to main page and ensure the element below renders. Hover over the branches and make sure the outline changes. Use tab to select the branches and make sure the outline changes (a11y check)
image

Try changing around the demo element in MainPage.jsx and make sure it does not break (under reasonable constraints)

Also LMK if you have design suggestions :)

### CODE REVIEW

File: front-end/src/components/IntentDiagram.test.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        expect(percentage).toBeInTheDocument();
+        expect(percentage.textContent).toBe(`${value}%`)
+        }
+    });
+
+    it("correctly displays 3 branches", () => {
+        props.branches = {
+            q1: 50,
+            q2: 20,
+            q3: 30,
```
Comment (williamyao27) [inline front-end/src/components/IntentDiagram.test.js:56]: This is more a general question related to testing, would there be any additional value in randomizing these percentages instead of hard-coding them into the test? I've seen some testing scripts do that, especially if the test isn't testing a specific edge case that requires specific values. It shouldn't make a difference for a simpler app like ours, just curious.

Comment (WeissSa) [reply front-end/src/components/IntentDiagram.test.js:56]: Randomizing could make sense if testing if a specific algorithm is applied correctly to them (i.e. we randomize the percentages, divide by 2, and then test to see if that is what is being displayed), but since this is 1 for 1, I think using hard-coded values should be fine. 

Essentially, I think both random and non-random values would be valid in this case.

**Review by KayleeLChan — APPROVED** [review]
I QA'ed and everything went as expected! Feels good, so I think it's going great 💃 💃 💃 

Some design suggestions:
- Using complimentary colours but at different saturations, everything is very bright right now, which isn't a great strain on the eyes.
- Love rounded corners on squares over ovals personally, plus I feel like that fits the VoiceFlow vibe so having a familiar vibe to theirs might be a good note.

**Comment by WeissSa** [comment]
Will update styles with suggestions and then merge 👍
